src/engine/Timer.ts

```diff
@@ -59,14 +70,37 @@ export class Timer {
     }
 
     this.id = Timer._MAX_ID++;
-    this.interval = interval || this.interval;
-    this.repeats = repeats || this.repeats;
+    if (!!randomRange){
+      if(randomRange[0] > randomRange[1]){
+        throw new Error('min value must be lower than max value for range');
+      } else if(!!random){
+        //We use the instance of ex.Random to generate the range
+        this.random = random;
+        
+        this.interval = interval + this.random.integer(randomRange[0], randomRange[1])
+      } else {
+        this.interval = interval + new Random().integer(randomRange[0], randomRange[1])
+      }
+      this.randomRange = randomRange;  
+    } else {
+      this.interval = interval;
+    }
 
+    this.repeats = repeats || this.repeats;
+    
     this._callbacks = [];
```

Comment:
```suggestion
    this._callbacks = [];
    this._baseInterval = this.interval = interval;
    if (!!randomRange){
      if(randomRange[0] > randomRange[1]){
        throw new Error('min value must be lower than max value for range');
      }
      //We use the instance of ex.Random to generate the range
      this.random = random ?? new Random();
      this.randomRange = randomRange;

      this.interval = this._generateRandomInterval();
      this.on(() => {
        this.interval = this._generateRandomInterval();
      });
    }
```

Reply:
We can take advantage of `this.random = random ?? new Random()` (this means if random is not passed in use a new Random) and the `_generateRandomInterval()` helper to simplify a bit